lib/triplex.ex

```diff
@@ -379,7 +379,13 @@ defmodule Triplex do
   end
 
   def to_prefix(tenant, nil), do: tenant
-  def to_prefix(tenant, prefix), do: "#{prefix}#{tenant}"
+
+  def to_prefix(tenant, prefix) do
+    cond do
+      !String.contains?(tenant, prefix) -> "#{prefix}#{tenant}"
+      true -> tenant
+    end
+  end
```

Comment:
Well, the idea is that `to_prefix("test", "test")` should still return `"testtest"`. And it would not solve the problem at all anyway, as `Triple.all()` do not call this function at all.

The change should be done here https://github.com/ateliware/triplex/blob/master/lib/triplex.ex#L293. Basically, add something like `|> Enum.map(&String.replace(&1, prefix, ""))` to that pipe would do the job, you just need to add the `prefix \\ config().tenant_prefix` param to the function

Reply:
Hi Kelvin,

My thinking is that you may want the full schema name when you call `Triplex.all()` (including the prefix if you have defined one). The issue you described happens when you then iterate over the list to run migrations so I thought testing for duplication would ensure that if the prefix exists it will not double up.

Have I got this wrong?

I am happy to make change as per above just want to make sure I have it straight.

Thanks,

Andrew

Reply:
Well, true, that would be one way to solve it, but the problem is that it still would be a mess if I wanted to make sure that the prefix is ALWAYS added to the tenant name.

Reply:
Ok, I will fix it up. Thanks